### ISSUE TITLE
Searching encrypted chats on web

### ISSUE THREAD

**Issue by chagai95** [vector-im/element-web#16483]
**Is your suggestion related to a problem? Please describe.**

Can not use Jitsi Screen Sharing From Electron App so was thinking of the Chrome app as an alternative and that would be missing there

**Describe the solution you'd like**
What are the plans for this?
Is it possible to enable somehow even though it will hurt security?
I understood deleted messages might also be affected from this or is this solved already?

**Describe alternatives you've considered**
Electron App

**Additional context**

**Comment by jryans** [comment]
@chagai95 The issue text doesn't match the title, could you revise it?

**Comment by chagai95** [comment]
Sorry @jryans I changed it a bit, I added a bit too much context I guess but the title does sum up the issue pretty well

**Comment by jryans** [comment]
At the moment, we don't have a clear path to resolve this. The desktop app leverages [Seshat](https://github.com/matrix-org/seshat) to index all your messages client side. 

Some day it might be possible to use it in the browser as well via WebAssembly or similar, but at the moment it's quite hard / impossible as Seshat requires threads, which are not well supported via Wasm at this time. 

We could potentially implement yet another, different, browser-specific indexing approach, but this would be a large investment.

Perhaps someone sponsoring this feature directly will be the fastest path forward, though even then it's unclear what the right solution would be technically.

**Comment by chagai95** [comment]
Thanks for explaining, unfortunately I still did not quite understand... I thought Electron is just the same as Web Element just as a separate chromium. What is the difference between what Electron does with chromium and Chrome for example? 

Is the key word here "native" components? You can not install node components on Chrome but you can pack them in Electron?

**Comment by t3chguy** [comment]
> Is the key word here "native" components? You can not install node components on Chrome but you can pack them in Electron?

Precisely, Seshat runs in the native side which can instantiate threads

**Comment by chagai95** [comment]
So there is not much hope for https://github.com/stoically/radical-native?

**Comment by kevincox** [comment]
I opened an issue against seshat since it seems like it is the primary blocker and it is unlikely that element switches indexer for the foreseeable future: https://github.com/matrix-org/seshat/issues/84

**Comment by novocaine** [comment]
Notes from @poljar on what's required for this

> the main blocker is tantivy not working on WASM, after that works seshat itself needs a bit of a rework, well we probably want to rewrite the public API to be async and probably use indexeddb for storage

**Comment by novocaine** [comment]
seshat issue: https://github.com/matrix-org/seshat/issues/84

**Comment by quininer** [comment]
Considering using [sqlite wasm](https://sqlite.org/wasm/doc/tip/about.md) instead of indexeddb? This requires browser to support [OPFS](https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API#origin_private_file_system), but it makes it easy to support search.

Firefox recently enabled this feature by default. https://github.com/mozilla/gecko-dev/commit/fd1047817ec99a8c625302c4c4b69ca0c75b1b5b

**Comment by t3chguy** [comment]
OPFS also isn't supported in Safari 15.1 which is within our support policy

**Comment by Malix-Labs** [comment]
If not using IndexedDB, I recommend [Turso](https://github.com/tursodatabase/turso) instead of [SQLite WASM](https://sqlite.org/wasm/doc/tip/about.md), and it has the benefit to be written in Rust too (like https://github.com/matrix-org/seshat), which might ease integration
